### PR TITLE
UVCmetrics role

### PR DESCRIPTION
**Pull request by zachmullen**

### CODE REVIEW

File: roles/uvcmetrics/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+
+    uvcdat_src_path: "/opt/uvcdat-src"
+    uvcdat_build_path: "/opt/uvcdat-build"
+    uvcdat_install_path: "/opt/uvcdat"
```
Comment (opadron) [inline roles/uvcmetrics/defaults/main.yml]: For the sake of consistency, let's default to installing under

```
/opt/uvcdat/{{ uvcdat_version }}
```

where `uvcdat_version` is also configurable (maybe default to "master").

Have a look at the [zookeeper-variables](https://github.com/opadron/gobig/blob/spark-role/roles/zookeeper-variables/tasks/main.yml) role (under my spark branch) for an example of what I've been doing with variables/directories/various options.

File: roles/uvcmetrics/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      -DCDAT_BUILD_OFFSCREEN=ON
+      -DCMAKE_INSTALL_PREFIX={{ uvcdat_install_path }}
+      {{ uvcdat_src_path }}
+    args:
+      chdir: "{{ uvcdat_build_path }}"
+
+  - name: uvcmetrics | uvcdat | build
+    command: make -j3
+    args:
+      chdir: "{{ uvcdat_build_path }}"
```
Comment (opadron) [inline roles/uvcmetrics/tasks/main.yml:56]: It's probably a good idea to detect how many cores you've got to work with.

[Here's](https://github.com/opadron/gobig/blob/spark-role/roles/mesos-install/tasks/main.yml#L149-L154) an example where I do just that (also in my spark branch).

Comment (danlamanna) [inline roles/uvcmetrics/tasks/main.yml:56]: Is this something we can use [`ansible_processor_cores`](http://docs.ansible.com/ansible/playbooks_variables.html#information-discovered-from-systems-facts) for?

Comment (opadron) [inline roles/uvcmetrics/tasks/main.yml:56]: Cool!  That looks way better than my hack. :+1:

**Comment by zachmullen** [comment]
Is this good to go now?

**Comment by opadron** [comment]
@zachmullen Could you add stand-alone playbooks for installing and uninstalling?  Otherwise, LGTM.

**Comment by zachmullen** [comment]
ping @opadron

**Pull request merged**